src/ripple/basics/base_uint.h

```diff
@@ -236,27 +233,26 @@ class base_uint
             {
                 if (auto const result = hexCharToUInt(*in++, shift, accum);
                     result != ParseResult::okay)
-                {
-                    ret.first = result;
-                    return ret;
-                }
+                    return Unexpected(result);
             }
-            *out++ = accum;
+            out[i++] = accum;
         }
-        return ret;
+        return out;
     }
 
     constexpr decltype(data_)
     parseFromStringViewThrows(std::string_view sv) noexcept(false)
     {
         auto const result = parseFromStringView(sv);
-        if (result.first == ParseResult::badLength)
-            Throw<std::invalid_argument>("invalid length for hex string");
+        if (!result)
+        {
+            if (result.error() == ParseResult::badLength)
+                Throw<std::invalid_argument>("invalid length for hex string");
 
-        if (result.first == ParseResult::badChar)
+            assert(result.error() == ParseResult::badChar);
```

Comment:
Why does `badChar` assert but `badLength` doesn't assert? Consider losing the `assert`. It's changing the behavior of debug builds, and the change doesn't involve the major focus of this PR - which is `expected`. I think it's better if PRs focus on one thing only.

Having said that, I only weakly object. If you feel strongly, then it's your call.

Reply:
The assert was intended to be a way to guarantee that all of the non-`okay` `ParseResult` values are covered.  We can see that by inspection now.  But in case another error code was added in the future the `assert` would let us know that the `Throw` had the wrong text.  But I don't feel strongly about it.  I'm fine with removing the `assert`.